### PR TITLE
SURF-508 Edits for checkboxes

### PR DESCRIPTION
**Pull request by asettle**
Closes: https://jira.rax.io/browse/SURF-508

### CODE REVIEW

**Comment by araiford** [comment]
@asettle @robb-romans 

Alex, Do you mind adding the two-column, responsive classes to the descriptions and figures for this doc?

I've got a PR currently pending you could reference.  It adds the responsive column markup to a couple docs.

Robb and Alex, I'm wondering how to document using the responsive column classes.  Should I update the components_template file in _posts with the responsive class markup?

File: _component_design/checkboxes.md [hunk 1]
```diff
@@ -17,150 +17,139 @@ last-modified: 2017-08-17
...
-      <button class="ui button ds-btn-med-sec">12<span class="smCaps"> PM</span> &ndash; 4<span class="smCaps"> PM</span></button>
-      <button class="ui button ds-btn-med-sec">4<span class="smCaps"> PM</span> &ndash; 8<span class="smCaps"> PM</span></button>
-      <button class="ui button ds-btn-med-sec">8<span class="smCaps"> PM</span> &ndash; 12<span class="smCaps"> AM</span></button>
-    </div>
-    <div class="ui error message">
-      There is an error with the response to this question.
-    </div>
-  </div>
-</div>
+Checkboxes are used to change settings or bulk item selection. Checkboxes
```
Comment (robb-romans) [inline _component_design/checkboxes.md]: Use checkboxes to change...

File: _component_design/checkboxes.md [hunk 2]
```diff
@@ -17,150 +17,139 @@ last-modified: 2017-08-17
...
+allow the user to select zero, one, or several items.
+
+{% figure [caption:"Default image"] [class:"image bg-light border"] %}
+ ![]({{site.url}}/assets/images/components/content-areas/checkboxes/checkbox-hero-image.svg){:width="80%"}
+ {% endfigure %}
+
+## When to use checkboxes
+
+* Using a checkable input to allows the user to select any number of options;
```
Comment (robb-romans) [inline _component_design/checkboxes.md]: You can use checkable input to select...

File: _component_design/checkboxes.md [hunk 3]
```diff
@@ -17,150 +17,139 @@ last-modified: 2017-08-17
...
+  including zero, one, or several.
+* Use when selecting multiple items from a set, it is important for the
+  user to view all options at once. If viewing items side-by-side is not
```
Comment (robb-romans) [inline _component_design/checkboxes.md]: side by side

File: _component_design/checkboxes.md [hunk 4]
```diff
@@ -17,150 +17,139 @@ last-modified: 2017-08-17
...
+  important, consider a drop-down as it uses less space.
+* A stand-alone checkbox is used for a single option that the user can
```
Comment (robb-romans) [inline _component_design/checkboxes.md]: Recommend active voice as much as possible, throughout.

File: _component_design/checkboxes.md [hunk 5]
```diff
@@ -17,150 +17,139 @@ last-modified: 2017-08-17
...
+  turn on or off.
+* If there are multiple choices and only one option can be selected, a
+  radio button or drop-down list should be used.
+
+## Best practices for checkboxes
+
+* Each input is independent of all other input in the list and checking one
+  box does not uncheck the others.
+* A singular input may be used to enable or disable a feature, or it
```
Comment (robb-romans) [inline _component_design/checkboxes.md]: s/may/can/g

File: _component_design/checkboxes.md [hunk 6]
```diff
@@ -17,150 +17,139 @@ last-modified: 2017-08-17
...
+  may be used in place of two yes and no radio buttons.
+* Use checkboxes and radio buttons only to change settings, not as action
+  buttons that trigger another process or workflow.
+* In most cases, checkboxes default to having none of the options
+  selected. However, if Rackspace has a recommended choice, that may be
+  used as the default.
```
Comment (robb-romans) [inline _component_design/checkboxes.md]: Typically, checkboxes default to having no options selected. However, if Rackspace recommended an option, make that the default.

File: _component_design/checkboxes.md [hunk 7]
```diff
@@ -17,150 +17,139 @@ last-modified: 2017-08-17
...
+  check a particular box, and what will happen if they leave it unchecked.
+  If you are unable to do this, we recommend using two radio buttons: one
+  for having the feature on, and one for having it off. Ensure you write
+  clear labels for each of the two cases.
+
+### Presentation for checkboxes
+
+* If possible, use checkboxes rather than multi-select drop-down menus.
+  Checkboxes have lower cognitive load because they make all options
+  permanently visible so that users can easily compare them. Checkboxes
```
Comment (robb-romans) [inline _component_design/checkboxes.md]: rm "permanently"

File: _component_design/checkboxes.md [hunk 8]
```diff
@@ -17,150 +17,139 @@ last-modified: 2017-08-17
...
+
+A checkbox group consists of a Section Label, followed pairs of checkboxes and checkbox labels.
+
+{% figure [caption:"Checkbox group composition and specifications"] [class:"image bg-light border"] %}
+ ![]({{site.url}}/assets/images/components/content-areas/checkboxes/checkbox-default.svg){:width="80%"}
+ {% endfigure %}
+
+## Checkbox group specifications
+
+* Visually present groups of choices as groups, and clearly separates the
```
Comment (robb-romans) [inline _component_design/checkboxes.md]: groups, and clearly separate the

File: _component_design/checkboxes.md [hunk 9]
```diff
@@ -17,150 +17,139 @@ last-modified: 2017-08-17
...
+  choices from other groups on the same page.
+* Use sub-heads to break up a long list of checkboxes into logical groups.
+  This makes the choices faster to scan and easier to understand. The risk
+  is that users might view each sub-group as a separate set of options, but
+  this is not necessarily fatal for checkboxes, each box is an independent
```
Comment (robb-romans) [inline _component_design/checkboxes.md]: Change to something like: "With sub-heads, users might misunderstand each sub-group as a separate set of options. With checkboxes, each box is an independent choice."

File: _component_design/checkboxes.md [hunk 10]
```diff
@@ -17,150 +17,139 @@ last-modified: 2017-08-17
...
+* De-selected
+* Indeterminate
+
+{% figure [caption:"Checkbox states"] [class:"image bg-light border"] %}
+ ![]({{site.url}}/assets/images/components/content-areas/checkboxes/checkbox-states-image.svg){:width="80%"}
+ {% endfigure %}
+
+## Required group
+
+A red asterisk may be used to designate that a selection is required for a
```
Comment (robb-romans) [inline _component_design/checkboxes.md]: Use a red asterisk to...

File: _component_design/checkboxes.md [hunk 11]
```diff
@@ -17,150 +17,139 @@ last-modified: 2017-08-17
...
+group of checkboxes.
+The label should remain in its natural vertical alignment. The asterisk
+position floats to the left not in the vertical alignment with the text.
```
Comment (robb-romans) [inline _component_design/checkboxes.md]: Confused by this sentence.

Comment (asettle) [reply _component_design/checkboxes.md]: Yeah me too. @technabors ? Halp

**Pull request merged**